src/services/completions.ts

```diff
@@ -371,6 +381,8 @@ namespace ts.Completions {
                 }
             }
 
+            result = result.filter(entry => entry.name.indexOf(baseName) >= 0);

```

Comment:
I've only checked vs and vs code. It appears vs doesn't do any additional filtering, but vs code does (namely, it repeats exactly the same fuzzy matching I've added).

In order to offer a consistent experience, and so typing additional characters refines completions, I added the filtering.

Changed the checking to occur earlier, but I worry slightly that we might be violating DRY.